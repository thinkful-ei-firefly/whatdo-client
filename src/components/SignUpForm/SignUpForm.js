import React from "react";
import AuthApiService from "../../services/auth-api-service";
import { Link } from "react-router-dom";
import TokenService from "../../services/token-service";
import { UserContext } from "../../contexts/UserContext";

export default class SignUpForm extends React.Component {
  static contextType = UserContext;
  state = { error: null };

  handleSubmit = async e => {
    e.preventDefault();
    const { username, password, nickname, email } = e.target;
    const newUser = {
      username: username.value,
      password: password.value,
      nickname: nickname.value,
      email: email.value
    };

    try {
      const createdUser = await AuthApiService.createAccount(newUser);
      username.value = "";
      password.value = "";
      nickname.value = "";

      // Auto login after account creation
      const token = await AuthApiService.login({
        username: createdUser.username,
        password: newUser.password
      });
      TokenService.saveAuthToken(token.authToken);
      this.context.setUser({ username: newUser.username });
      // redirect to eventsPage
      // should update this to show a 'Success' message on next page
      this.props.onSuccessfulSignUp();
    } catch (err) {
      this.setState({ error: err.error });
    }
  };

  render() {
    const { error } = this.state;
    return (
      <div className="signup-form">
        <form className="SignUpForm" onSubmit={this.handleSubmit}>
          <div className="alert">{error && <p>{error}</p>}</div>
          <h1>Sign Up</h1>
          <input
            className="txtb"
            autoComplete="off"
            type="text"
            id="username"
            name="username"
            placeholder="Username"
            required
          />
          <input
            className="txtb"
            autoComplete="off"
            type="text"
            id="nickname"
            name="nickname"
            placeholder="Nickname"
            required
          />
          <input
            className="txtb"
            autoComplete="off"
            type="email"
            id="email"
            name="email"
            placeholder="Email"
            required
          />
          <input
            className="txtb"
            type="password"
            id="password"
            name="password"
            placeholder="Password"
            required
          />
          <input className="signup-btn" type="submit" value="Sign Up!" />
          <Link to="/signIn" className="alreadyHaveAcc">
            Already Have one ?
          </Link>
        </form>
      </div>
    );
  }
}
