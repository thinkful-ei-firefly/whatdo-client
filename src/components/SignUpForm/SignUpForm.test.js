import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import SignUpForm from './SignUpForm'
import renderer from 'react-test-renderer'

describe('<SignUpForm />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <SignUpForm />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <SignUpForm />
      </BrowserRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
