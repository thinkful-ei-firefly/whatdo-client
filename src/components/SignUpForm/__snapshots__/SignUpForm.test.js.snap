// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> renders correctly 1`] = `
<div
  className="signup-form"
>
  <form
    className="SignUpForm"
    onSubmit={[Function]}
  >
    <div
      className="alert"
    />
    <h1>
      Sign Up
    </h1>
    <input
      autoComplete="off"
      className="txtb"
      id="username"
      name="username"
      placeholder="Username"
      required={true}
      type="text"
    />
    <input
      autoComplete="off"
      className="txtb"
      id="nickname"
      name="nickname"
      placeholder="Nickname"
      required={true}
      type="text"
    />
    <input
      autoComplete="off"
      className="txtb"
      id="email"
      name="email"
      placeholder="Email"
      required={true}
      type="email"
    />
    <input
      className="txtb"
      id="password"
      name="password"
      placeholder="Password"
      required={true}
      type="password"
    />
    <input
      className="signup-btn"
      type="submit"
      value="Sign Up!"
    />
    <a
      className="alreadyHaveAcc"
      href="/"
      onClick={[Function]}
    >
      Already Have one ?
    </a>
  </form>
</div>
`;
